src/os/bluestore/BlueStore.cc

```diff
@@ -6028,21 +6028,25 @@ int BlueStore::_do_alloc_write(
 	csum_order = std::min(wctx->csum_order, ctz(l->length()));
       }
     }
-    while (final_length > 0) {
+
+    int count = 0;
+    std::vector<AllocExtent> extents = 
+                std::vector<AllocExtent>(final_length / min_alloc_size);
+
+    int r = alloc->alloc_extents(final_length, min_alloc_size, max_alloc_size,
+                                 hint, &extents, &count);
+    need = 0;
+    assert(r == 0);
+    for (int i = 0; i < count; i++) {
       bluestore_pextent_t e;
-      uint32_t l;
-      uint64_t want = max_alloc_size ? MIN(final_length, max_alloc_size) : final_length;
-      int r = alloc->allocate(want, min_alloc_size, hint,
-			      &e.offset, &l);
-      assert(r == 0);
-      need -= l;
-      e.length = l;
+      e.offset = extents[i].offset;
+      e.length = extents[i].length;
+
       txc->allocated.insert(e.offset, e.length);
       txc->statfs_delta.allocated() += e.length;
       b->blob.extents.push_back(e);
-      final_length -= e.length;
-      hint = e.end();
```

Comment:
Update hint for backward compatibility? Perhaps stupidAllocator still needs it...